optuna/visualization/_utils.py

```diff
@@ -179,3 +180,27 @@ def _target(t: FrozenTrial) -> float:
 
 def _is_reverse_scale(study: Study, target: Optional[Callable[[FrozenTrial], float]]) -> bool:
     return target is not None or study.direction == StudyDirection.MINIMIZE
+
+
+def _make_json_compatible(value: Any) -> Any:
+    try:
+        json.dumps(value)
+        return value
+    except TypeError:
+        # The value can't be converted to JSON directly, so return a string representation.
+        return str(value)
+
+
+def _make_hovertext(trial: FrozenTrial) -> str:
```

Comment:
I'm not fully sure, but how about removing the `_` prefix and exposing this function?
Currently, only `is_available` is included in `__all__`, but `_make_hovertext` is also used in the other files such as `_pareto_front.py` and `_timeline.py`.

CC @HideakiImamura 

```suggestion
def make_hovertext(trial: FrozenTrial) -> str:
```

Reply:
Ah, users may use `is_available`, but they will not use `_make_hovertext`. We should keep it private.